docs/python/doxygenlib/cdWriterDocstring.py

```diff
@@ -687,49 +699,89 @@ def __getFullDoc(self, pyname, pyobj, doxy):
         # make the doxy element static if it is tagged as such
         if ATTR_STATIC_METHOD in doxy.doc['tags']:
             doxy.static = 'yes'
-        
+
         lines = self.__getShortDescription(pyname, pyobj, doxy)
         if doxy.isFunction() and type(pyobj) != property:
-            lines += self.__getSignatureDescription(pyname, pyobj, doxy)
-            lines.append('')
+            description = self.__getSignatureDescription(pyname, pyobj, doxy)
+            if description is not None:
+                lines += description
+                lines.append('')
         lines += self.__getDocumentation(pyname, pyobj, doxy)
         lines.append('')
         return lines
 
-    def __getOutputFormat(self, pypath, pyobj, overloads):
-        """Return the line that installs the docstring into the namespace."""
+    @classmethod
+    def __stripBoostSig(cls, doc):
+        def looksLikeBoostSig(l):
+            return bool(l and (l[0].isalnum() or l[0] == "_") and ' -> ' in l)
+
+        lines = doc.strip().splitlines()
+        if len(lines) and looksLikeBoostSig(lines[0]):
+            # boost signature has been prepended.  that means if there is
+            # an existing description provided in the C++ wrapper it will be
+            # indented.  strip the signature and dedent the description.
```

Comment:
I think an example before / after docstring (either real or made up) would greatly help understanding here.  I had to re-read this a few times before I thought I understood what's going on, as my initial assumption about what these would look like was wrong.